### DISCUSSION TITLE
taro-react真机上，点击事件有延迟

### DISCUSSION THREAD

**Discussion by tanghx** [NervJS/taro#9913] [Q & A]
taro-react在开发工具中是没有延迟的，但是在真机上点击事件会有延迟，而且还挺长。onClick，onTouchEnd，onTouchStart，onTap都试过了，都有延迟。用taro-vue试了下，也有延迟，但是没有react这么长，这个问题怎么解决。

**Comment by Chen-jj** [comment]
1. debug 一下这个函数（所有小程序事件的统一回调函数）

https://github.com/NervJS/taro/blob/762210351db9b28f550e09a80e3c22400f57e5c6/packages/taro-runtime/src/dom/event.ts#L114

看看回调触发时间。

2. 根据经验，页面结构复杂时，低端机也会有卡顿的感觉，可以参考[这里](https://docs.taro.zone/blog/2021-02-08-taro-jxpp#2-%E6%B8%B2%E6%9F%93%E6%80%A7%E8%83%BD%E4%BC%98%E5%8C%96)试试。

**Reply by Tanzuowei** [reply]
用custom-wrapper组件包裹一下会大量渲染节点处就好多了，这东西好多坑，vue组件用了v-if还不会自动销毁，只能通过控制key来销毁

**Comment by Class01192** [comment]
您好，请问有解决掉这个问题么，最近在做公司项目也遇到了，点击有明显延迟 taro-react。
